src/Tvheadend.cpp

```diff
@@ -2256,10 +2348,29 @@ void CTvheadend::ParseChannelDelete(htsmsg_t* msg)
   }
   Logger::Log(LogLevel::LEVEL_DEBUG, "delete channel %u", u32);
 
-  /* Erase */
+  /* We need to check and update providers. May be the deleted channel is the last channel with this provider */
+  uint32_t providerUid{0};
+  const auto it = m_channels.find(u32);
+  if (it != m_channels.cend())
+    providerUid = (*it).second.GetProviderUid();
+
+  /* Erase channel */
   m_channels.erase(u32);
   m_channelTuningPredictor.RemoveChannel(u32);
   TriggerChannelUpdate();
+
+  if (providerUid > 0)
```

Comment:
Should this be `providerUid != PVR_PROVIDER_INVALID_UID`?

Reply:
Yep. Thanks.

Reply:
Wait! PVR_PROVIDER_INVALID_UID is signed and defined as -1. providerUid is unsigned.

Reply:
Will change type accordingly.

Reply:
done